processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -426,4 +507,29 @@ public Row apply(Object input)
       }
     };
   }
+
+
+  /**
+   * This function checks the query for dimensions which can be optimized by applying the dimension extraction
+   * as the final step of the query instead of on every event.
+   * @param query The query to check for optimizations
+   * @param getPre Specify if we're looking for the pre or post (getDimension or getOutputName) labels for the dimension
+   * @return A set of dimensions which can be extracted at the last second upon query completion.
+   */
+  public static Set<String> extractionsToRewrite(GroupByQuery query, final boolean getPre)
```

Comment:
maybe it would make sense to return the set of DimensionSpecs directly to remove the need for a flag?